### PR TITLE
CADC-12678 - refactor to ensure function execution via blueprint happens only if values are used.

### PR DESCRIPTION
**Pull request by SharonGoliath**
Make some "reset" calls unnecessary. Added a more classes to fine-tune the blueprint handling.

### CODE REVIEW

File: cfht2caom2/instruments.py [hunk 1]
```diff
@@ -912,125 +947,87 @@ def _update_plane_provenance(self):
...
+            for attribute in bp._plan:
+                if attribute.startswith('Observation'):
+                    attribute_names = attribute.split('.')
+                    y = x.get(attribute_names[1], attribute_names[1])
+                    if hasattr(self._observation, y):
+                        attribute_ptr = getattr(self._observation, y)
+                        if len(attribute_names) > 2:
+                            z = x.get(attribute_names[2], attribute_names[2])
+                            if hasattr(attribute_ptr, z):
+                                attribute_ptr = getattr(attribute_ptr, z)
```
Comment (andamian) [inline cfht2caom2/instruments.py:969]: This is the only difference between `if` and `else`. I would put the next 3 lines outside the `if/else`

File: cfht2caom2/instruments.py [hunk 2]
```diff
@@ -880,10 +887,35 @@ def _semi_deep_copy_plane(
                     cc.copy_chunk(chunk)
                 )
 
+    def make_axes_consistent(self):
+        pass
```
Comment (andamian) [inline cfht2caom2/instruments.py]: I'm just wondering if these methods shouldn't throw a not-implemented exception to force subclasses to consider them. It's a design pattern which name escapes me right now...

**Review by andamian — APPROVED** [review]

File: cfht2caom2/instruments.py (hunk 2 above)
Comment (SharonGoliath) [reply cfht2caom2/instruments.py]: I'll updated the code where it didn't break unit tests to do this, and will do this going forward.

File: cfht2caom2/instruments.py (hunk 1 above)
Comment (SharonGoliath) [reply cfht2caom2/instruments.py:969]: Moving the next 3 lines outside the `if/else` means the `clear` + `set` calls always occur. These call currently do not occur if the `hasattr` call files.

**Pull request merged**